### DISCUSSION TITLE
Sofar dxoverdy dashboard occassionally shows values

### DISCUSSION THREAD

**Discussion by kiwibhoy** [cmcgerty/Sofar2mqtt#71] [Q&A]
This is a difficult issue to explain I have been trying for ages to get the Sofar dashboard to show values and occassionally it has shown values but never updated them, the files below give you a snapshot of what I mean.
[Sofardashboard with values– Home Assistant PDF.pdf](https://github.com/cmcgerty/Sofar2mqtt/files/13210338/Sofardashboard.with.values.Home.Assistant.PDF.pdf)
[Sofar – Home Assistant dashboard shows no values PDF.pdf](https://github.com/cmcgerty/Sofar2mqtt/files/13210340/Sofar.Home.Assistant.dashboard.shows.no.values.PDF.pdf)

![IMG_6288](https://github.com/cmcgerty/Sofar2mqtt/assets/65698336/3dc837f0-0621-4da7-bdf3-92245c5af39e)

It would appear that the mqtt express app shows values coming through from the inverter all the time.
I have checked all connections even going to the bother of making up a vero board and soldering all connections and getting the same result.
The fact that mqtt express is showing values at all times is making me think I had issues in the way I loaded the sofar dashboard so I put on another instance of HA and installed the sofar dashboard again  and got the same result I then added the BME280 stuff and displayed the BME280 values on the NodeRed dashboard successfully.
I have used both a D1 Mini and a NodeMCU esp8266 connected to a RS485 to TTL board, with the same results.
I have checked the integrety of the network, although the fact that mqtt express always shows values coming through would point to the network being OK.
As I said I found this a difficult issue to explain I just hope someone point me in the right direction to making this work.

**Comment by kiwibhoy** [comment]
Additional information:
I have now realised that I can add the mqtt "explorer" (sorry I was calling it "express") as an addon to HA which I have done and some time after that I had explorer showing all values. In my excitment I forgot to do a screenshot and when I went to the Sofar dashboard which showed all the values and then back to the mqtt page all the values had disappeared.  Just thought it might be a clue to what is happening.

**Comment by cliffdude** [comment]
Looks like a problem with how you setup your MQTT entities.
The ESP does not sent all values in every message so what you see in MQTT
explorer is normal, it can only send what it has received and decoded from
the RS485 and I have to assume that the inverter does not send all data
every time.
There is code for HA in the repository, I would suggest creating sensors
with that code and testing again. Once you have the first valid message the
value of the sensor will persist in HA until the next MQTT message arrives
with a new value for that variable.

**Reply by kiwibhoy** [reply]
Thanks for your help, I assume you are talking about the ymal config. file.
I am using this config. file at the moment:


# Loads default set of integrations. Do not remove.
default_config:

# Load frontend themes from the themes folder
frontend:
  themes: !include_dir_merge_named themes
  extra_module_url:
    - /hacsfiles/lovelace-card-mod/card-mod.js


# Text to speech
tts:
  - platform: google_translate

automation: !include automations.yaml
script: !include scripts.yaml
scene: !include scenes.yaml






#MQTT
mqtt:
  button:
# House Buttons
    - unique_id: sofar_set_standby
      command_topic: "Sofar2mqtt/set/standby"
      name: "SOFAR Set Standby"
      payload_press: "true"
      qos: 0
      retain: false
      entity_category: "config"
      icon: "mdi:power-standby"
      
    - unique_id: sofar_set_auto
      command_topic: "Sofar2mqtt/set/auto"
      name: "SOFAR Set Auto"
      command_template: "true"
      qos: 0
      retain: false
      entity_category: "config"
      icon: "mdi:lightbulb-auto"
      
    - unique_id: sofar_set_battery_save
      command_topic: "Sofar2mqtt/set/auto"
      name: "SOFAR Set Battery Save"
      command_template: "battery_save"
      qos: 0
      retain: false
      entity_category: "config"
      icon: "mdi:lightbulb-auto"

    - unique_id: sofar_set_charge
      command_topic: "Sofar2mqtt/set/charge"
      name: "SOFAR Set Charge"
      command_template: "{{ states('input_number.input_house_charge_watts')|int }}"
      qos: 0
      retain: false
      entity_category: "config"
      icon: "mdi:battery-positive"

    - unique_id: sofar_set_discharge
      command_topic: "Sofar2mqtt/set/discharge"
      name: "SOFAR Set Discharge"
      command_template: "{{ states('input_number.input_house_discharge_watts')|int }}"
      qos: 0
      retain: false
      entity_category: "config"
      icon: "mdi:battery-negative"




#MQTT Sensors
  sensor:
#House Sensors
    - device_class: "temperature"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:thermometer"
      name: "SOFAR Battery Temperature"
      unique_id: "sofar_battery_temp"
      unit_of_measurement: "°C"
      state_topic: "sofar/battery_temp"
      value_template: "{{ value | float }}"

    - state_class: "measurement"
      force_update: "true"
      icon: "mdi:information-variant"
      name: "SOFAR Inverter State"
      unique_id: "sofar_inverter_state"
      state_topic: "sofar/inverter_state"

    - device_class: "voltage"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:alpha-v"
      name: "SOFAR Grid Voltage"
      unique_id: "sofar_grid_voltage"
      unit_of_measurement: "V"
      state_topic: "sofar/grid_voltage"
      value_template: "{{ value | float }}"

    - device_class: "current"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:alpha-a"
      name: "SOFAR Grid Current"
      unique_id: "sofar_grid_current"
      unit_of_measurement: "A"
      state_topic: "sofar/grid_current"
      value_template: "{{ value | float }}"

    - device_class: "frequency"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:sine-wave"
      name: "SOFAR Grid Frequency"
      unique_id: "sofar_grid_freq"
      unit_of_measurement: "Hz"
      state_topic: "sofar/grid_freq"
      value_template: "{{ value | float }}"

    - device_class: "power"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:battery-charging-100"
      name: "SOFAR Battery Power"
      unique_id: "sofar_battery_power"
      unit_of_measurement: "W"
      state_topic: "sofar/battery_power"
      value_template: "{{ value | float }}"

    - device_class: "voltage"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:alpha-v"
      name: "SOFAR Battery Voltage"
      unique_id: "sofar_battery_voltage"
      unit_of_measurement: "V"
      state_topic: "sofar/battery_voltage"
      value_template: "{{ value | float }}"

    - device_class: "current"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:alpha-a"
      name: "SOFAR Battery Current"
      unique_id: "sofar_battery_current"
      unit_of_measurement: "A"
      state_topic: "sofar/battery_current"
      value_template: "{{ value | float }}"

    - device_class: "battery"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:battery"
      name: "SOFAR Battery SOC"
      unique_id: "sofar_batterySOC"
      unit_of_measurement: "%"
      state_topic: "sofar/batterySOC"
      value_template: "{{ value | float }}"

    - state_class: "measurement"
      force_update: "true"
      icon: "mdi:battery-sync"
      name: "SOFAR Battery Cycles"
      unique_id: "sofar_battery_cycles"
      state_topic: "sofar/battery_cycles"
      value_template: "{{ value | float }}"

    - device_class: "power"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:transmission-tower"
      name: "SOFAR Grid Power"
      unique_id: "sofar_grid_power"
      unit_of_measurement: "W"
      state_topic: "sofar/grid_power"
      value_template: "{{ value | float }}"

    - device_class: "power"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:home-import-outline"
      name: "SOFAR Load"
      unique_id: "sofar_consumption"
      unit_of_measurement: "W"
      state_topic: "sofar/consumption"
      value_template: "{{ value | float }}"

    - device_class: "power"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:solar-power"
      name: "SOFAR Solar PV Generation"
      unique_id: "sofar_solarPV"
      unit_of_measurement: "W"
      state_topic: "sofar/solarPV"
      value_template: "{{ value | float }}"

    - device_class: "power"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:solar-power"
      name: "SOFAR Solar PV Generation Array 1"
      unique_id: "sofar_solarPVArray1"
      unit_of_measurement: "W"
      state_topic: "sofar/Solarpv1"
      value_template: "{{ value | float }}"

    - device_class: "power"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:solar-power"
      name: "SOFAR Solar PV Generation Array 2"
      unique_id: "sofar_solarPVArray2"
      unit_of_measurement: "W"
      state_topic: "sofar/Solarpv2"
      value_template: "{{ value | float }}"

    - device_class: "power"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:arrow-decision"
      name: "SOFAR System IO Power"
      unique_id: "sofar_system_io_power"
      unit_of_measurement: "W"
      state_topic: "sofar/systemIO_power"
      value_template: "{{ value | float }}"

    - device_class: "power"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:solar-power"
      name: "SOFAR Solar PV Generation"
      unique_id: "sofar_solarPV"
      unit_of_measurement: "W"
      state_topic: "sofar/solarPV"
      value_template: "{{ value | float }}"

    - device_class: "energy"
      state_class: "total_increasing"
      force_update: "true"
      icon: "mdi:solar-power"
      name: "SOFAR Today's PV Generation"
      unique_id: "sofar_today_generation"
      unit_of_measurement: "kWh"
      state_topic: "sofar/today_generation"
      value_template: "{{ value | float }}"

    - device_class: "energy"
      state_class: "total_increasing"
      force_update: "true"
      icon: "mdi:transmission-tower-export"
      name: "SOFAR Today's Feed-In"
      unique_id: "sofar_today_export"
      unit_of_measurement: "kWh"
      state_topic: "sofar/today_export"  
      value_template: "{{ value | float }}"

    - device_class: "energy"
      state_class: "total_increasing"
      force_update: "true"
      icon: "mdi:transmission-tower-import"
      name: "SOFAR Today's Grid Consumption"
      unique_id: "sofar_today_import"
      unit_of_measurement: "kWh"
      state_topic: "sofar/today_import"  
      value_template: "{{ value | float }}"

    - device_class: "energy"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:home-import-outline"
      name: "SOFAR Today's Load"
      unique_id: "sofar_today_consumption"
      unit_of_measurement: "kWh"
      state_topic: "sofar/today_consumption"  
      value_template: "{{ value | float }}"

    - device_class: "temperature"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:thermometer"
      name: "SOFAR Inverter Temperature"
      unique_id: "sofar_inverter_temp"
      unit_of_measurement: "°C"
      state_topic: "sofar/inverter_temp"
      value_template: "{{ value | float }}"

    - device_class: "temperature"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:thermometer"
      name: "SOFAR Inverter Heatsink Temperature"
      unique_id: "sofar_inverter_heatsink_temp"
      unit_of_measurement: "°C"
      state_topic: "sofar/inverter_heatsink_temp"
      value_template: "{{ value | float }}"

    - device_class: "current"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:alpha-a"
      name: "SOFAR PV Current"
      unique_id: "sofar_solarPVAmps"
      unit_of_measurement: "A"
      state_topic: "sofar/solarPVAmps"
      value_template: "{{ value | float }}"

    - device_class: "monetary"
      state_class: "total_increasing"
      force_update: "true"
      icon: "mdi:currency-gbp"
      name: "SOFAR Today's Total Generation Earned"
      unique_id: "sofar_today_total_generation_earned"
      unit_of_measurement: "£"
      state_topic: "sofar/today_generation"
      value_template: "{{(value | float * 0.9345) | round(2)}}"

    - device_class: "monetary"
      state_class: "total_increasing"
      force_update: "true"
      icon: "mdi:currency-gbp"
      name: "SOFAR Today's PV Generation Earned"
      unique_id: "sofar_today_generation_earned"
      unit_of_measurement: "£"
      state_topic: "sofar/today_generation"
      value_template: "{{(value | float * 0.623) | round(2)}}"

    - device_class: "monetary"
      state_class: "total_increasing"
      force_update: "true"
      icon: "mdi:currency-gbp"
      name: "SOFAR Today's Feed-In Earned"
      unique_id: "sofar_today_export_earned"
      unit_of_measurement: "£"
      state_topic: "sofar/today_generation"  
      value_template: "{{(value | float * 0.3115) | round(2)}}"

    - device_class: "monetary"
      state_class: "total_increasing"
      force_update: "true"
      icon: "mdi:currency-gbp"
      name: "SOFAR Today's Grid Consumption Cost"
      unique_id: "sofar_today_import_cost"
      unit_of_measurement: "£"
      state_topic: "sofar/today_import" 
      value_template: "{{(value | float * 0.16) | round(2)}}"

    - device_class: "power"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:transmission-tower-import"
      name: "SOFAR Grid Power Importing"
      unique_id: "sofar_importing_grid_power"
      unit_of_measurement: "W"
      state_topic: "sofar/importing_grid_power"
      value_template: "{{ value | float }}"

    - device_class: "power"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:transmission-tower-export"
      name: "SOFAR Grid Power Exporting"
      unique_id: "sofar_exporting_grid_power"
      unit_of_measurement: "W"
      state_topic: "sofar/exporting_grid_power"
      value_template: "{{ value | float }}"

    - device_class: "power"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:transmission-tower"
      name: "SOFAR Grid Power Absolute"
      unique_id: "sofar_absolute_grid_power"
      unit_of_measurement: "W"
      state_topic: "sofar/absolute_grid_power"
      value_template: "{{ value | float }}"

    - state_class: "measurement"
      force_update: "true"
      icon: "mdi:transmission-tower"
      name: "SOFAR Grid Status"
      unique_id: "sofar_grid_status"
      state_topic: "sofar/grid_status"

    - device_class: "power"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:battery-arrow-up"
      name: "SOFAR Battery Power Charging"
      unique_id: "sofar_charging_battery_power"
      unit_of_measurement: "W"
      state_topic: "sofar/charging_battery_power"
      value_template: "{{ value | float }}"

    - device_class: "power"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:battery-arrow-down"
      name: "SOFAR Battery Power Discharging"
      unique_id: "sofar_discharging_battery_power"
      unit_of_measurement: "W"
      state_topic: "sofar/discharging_battery_power"
      value_template: "{{ value | float }}"

    - device_class: "power"
      state_class: "measurement"
      force_update: "true"
      icon: "mdi:battery-charging"
      name: "SOFAR Battery Power Absolute"
      unique_id: "sofar_absolute_battery_power"
      unit_of_measurement: "W"
      state_topic: "sofar/absolute_battery_power"
      value_template: "{{ value | float }}"

    - state_class: "measurement"
      force_update: "true"
      icon: "mdi:battery-sync"
      name: "SOFAR Battery Status"
      unique_id: "sofar_battery_status"
      state_topic: "sofar/battery_status"




#Sensors
sensor:
#House Helpers
  - platform: integration
    method: left
    unit_time: h
    unit_prefix: k
    source: sensor.sofar_battery_power_charging
    name: "SOFAR Today's Total Battery Charge"
    unique_id: "sofar_today_battery_charge"
    round: 2

  - platform: integration
    method: left
    unit_time: h
    unit_prefix: k
    source: sensor.sofar_battery_power_discharging
    name: "SOFAR Today's Total Battery Discharge"
    unique_id: "sofar_today_battery_discharge"
    round: 2


#House Utility Meters
utility_meter:
  utility_meter_daily_sofar_battery_charge:
    source: sensor.sofar_today_s_total_battery_charge
    cycle: daily
    name: "SOFAR Utility Meter - Daily Total Battery Charge"
    unique_id: "sofar_utility_meter_daily_total_battery_charge"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_daily_sofar_battery_discharge:
    source: sensor.sofar_today_s_total_battery_discharge
    cycle: daily
    name: "SOFAR Utility Meter - Daily Total Battery Discharge"
    unique_id: "sofar_utility_meter_daily_total_battery_discharge"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_daily_sofar_feed_in:
    source: sensor.sofar_today_s_feed_in
    cycle: daily
    name: "SOFAR Utility Meter - Daily Feed-In"
    unique_id: "sofar_utility_meter_daily_feed_in"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_daily_sofar_grid_consumption:
    source: sensor.sofar_today_s_grid_consumption
    cycle: daily
    name: "SOFAR Utility Meter - Daily Grid Consumption"
    unique_id: "sofar_utility_meter_daily_grid_consumption"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_daily_sofar_pv_generation:
    source: sensor.sofar_today_s_pv_generation
    cycle: daily
    name: "SOFAR Utility Meter - Daily PV Generation"
    unique_id: "sofar_utility_meter_daily_pv_generation"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_daily_sofar_load:
    source: sensor.sofar_today_s_load
    cycle: daily
    name: "SOFAR Utility Meter - Daily Load"
    unique_id: "sofar_utility_meter_daily_load"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_daily_sofar_feed_in_earned:
    source: sensor.sofar_today_s_feed_in_earned
    cycle: daily
    name: "SOFAR Utility Meter - Daily Feed-In Earned"
    unique_id: "sofar_utility_meter_daily_feed_in_earned"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_daily_sofar_grid_consumption_cost:
    source: sensor.sofar_today_s_grid_consumption_cost
    cycle: daily
    name: "SOFAR Utility Meter - Daily Grid Consumption Cost"
    unique_id: "sofar_utility_meter_daily_grid_consumption_cost"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_daily_sofar_pv_generation_earned:
    source: sensor.sofar_today_s_pv_generation_earned
    cycle: daily
    name: "SOFAR Utility Meter - Daily PV Generation Earned"
    unique_id: "sofar_utility_meter_daily_pv_generation_earned"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_daily_sofar_total_generation_earned:
    source: sensor.sofar_today_s_total_generation_earned
    cycle: daily
    name: "SOFAR Utility Meter - Daily Total Generation Earned"
    unique_id: "sofar_utility_meter_daily_total_generation_earned"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_monthly_sofar_battery_charge:
    source: sensor.sofar_today_s_total_battery_charge
    cycle: monthly
    name: "SOFAR Utility Meter - Monthly Total Battery Charge"
    unique_id: "sofar_utility_meter_monthly_total_battery_charge"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_monthly_sofar_battery_discharge:
    source: sensor.sofar_today_s_total_battery_discharge
    cycle: monthly
    name: "SOFAR Utility Meter - Monthly Total Battery Discharge"
    unique_id: "sofar_utility_meter_monthly_total_battery_discharge"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_monthly_sofar_feed_in:
    source: sensor.sofar_today_s_feed_in
    cycle: monthly
    name: "SOFAR Utility Meter - Monthly Feed-In"
    unique_id: "sofar_utility_meter_monthly_feed_in"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_monthly_sofar_grid_consumption:
    source: sensor.sofar_today_s_grid_consumption
    cycle: monthly
    name: "SOFAR Utility Meter - Monthly Grid Consumption"
    unique_id: "sofar_utility_meter_monthly_grid_consumption"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_monthly_sofar_pv_generation:
    source: sensor.sofar_today_s_pv_generation
    cycle: monthly
    name: "SOFAR Utility Meter - Monthly PV Generation"
    unique_id: "sofar_utility_meter_monthly_pv_generation"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_monthly_sofar_load:
    source: sensor.sofar_today_s_load
    cycle: monthly
    name: "SOFAR Utility Meter - Monthly Load"
    unique_id: "sofar_utility_meter_monthly_load"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_monthly_sofar_feed_in_earned:
    source: sensor.sofar_today_s_feed_in_earned
    cycle: monthly
    name: "SOFAR Utility Meter - Monthly Feed-In Earned"
    unique_id: "sofar_utility_meter_monthly_feed_in_earned"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_monthly_sofar_grid_consumption_cost:
    source: sensor.sofar_today_s_grid_consumption_cost
    cycle: monthly
    name: "SOFAR Utility Meter - Monthly Grid Consumption Cost"
    unique_id: "sofar_utility_meter_monthly_grid_consumption_cost"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_monthly_sofar_pv_generation_earned:
    source: sensor.sofar_today_s_pv_generation_earned
    cycle: monthly
    name: "SOFAR Utility Meter - Monthly PV Generation Earned"
    unique_id: "sofar_utility_meter_monthly_pv_generation_earned"
    offset: 0
    delta_values: false
    net_consumption: false
  utility_meter_monthly_sofar_total_generation_earned:
    source: sensor.sofar_today_s_total_generation_earned
    cycle: monthly
    name: "SOFAR Utility Meter - Monthly Total Generation Earned"
    unique_id: "sofar_utility_meter_monthly_total_generation_earned"
    offset: 0
    delta_values: false
    net_consumption: false





input_number:
  input_house_charge_watts:
    name: "Input House Charge Watts"
    initial: 3000
    min: 0
    max: 3000
    step: 100
    mode: slider
    icon: "mdi:battery-positive"
  input_house_discharge_watts:
    name: "Input House Discharge Watts"
    initial: 3000
    min: 0
    max: 3000
    step: 100
    mode: slider
    icon: "mdi:battery-negative"




#Recorder
# Pointless keeping some values over time
# So looking to keep those we want to chart at any sort of frequency
# Battery charging/discharging, SOC, importing/exporting, load and PV generation
# And also the utility meters as they are supposedly going to keep us aggregate figures by day

recorder:
  exclude:
    entities:
      - sensor.sofar_battery_current
      - sensor.sofar_battery_cycles
      - sensor.sofar_battery_power
      - sensor.sofar_battery_power_absolute
      - sensor.sofar_battery_power_charging
      - sensor.sofar_battery_power_discharging
#      - sensor.sofar_battery_soc
      - sensor.sofar_battery_status
      - sensor.sofar_battery_temperature
      - sensor.sofar_battery_voltage
      - sensor.sofar_grid_current
      - sensor.sofar_grid_frequency
      - sensor.sofar_grid_power
      - sensor.sofar_absolute_grid_power
#      - sensor.sofar_exporting_grid_power
#      - sensor.sofar_importing_grid_power
      - sensor.sofar_grid_status
#      - sensor.sofar_grid_voltage
      - sensor.sofar_inverter_heatsink_temperature
#      - sensor.sofar_inverter_state
      - sensor.sofar_inverter_temperature
#      - sensor.sofar_load
      - sensor.sofar_pv_current
#      - sensor.sofar_solar_pv_generation
      - sensor.sofar_today_s_battery_charge
      - sensor.sofar_today_s_battery_discharge
      - sensor.sofar_today_s_feed_in
      - sensor.sofar_today_s_feed_in_earned
      - sensor.sofar_today_s_grid_consumption
      - sensor.sofar_today_s_grid_consumption_cost
      - sensor.sofar_today_s_load
      - sensor.sofar_today_s_pv_generation
      - sensor.sofar_today_s_pv_generation_earned
#      - sensor.sofar_today_s_total_generation_earned
#      - sensor.sofar_utility_meter_daily_battery_charge
#      - sensor.sofar_utility_meter_daily_battery_discharge
#      - sensor.sofar_utility_meter_daily_feed_in
#      - sensor.sofar_utility_meter_daily_feed_in_earned
#      - sensor.sofar_utility_meter_daily_grid_consumption
#      - sensor.sofar_utility_meter_daily_grid_consumption_cost
#      - sensor.sofar_utility_meter_daily_load
#      - sensor.sofar_utility_meter_daily_pv_generation
#      - sensor.sofar_utility_meter_daily_pv_generation_earned
#      - sensor.sofar_utility_meter_daily_total_generation_earned

#      - input_number.input_house_charge_watts
#      - input_number.input_house_discharge_watts

If this isn't the sensor file you meant could you point me to the file you advise me to use.
In case I haven't said before my inverter is a HYD 5000 ES.


I also managed to get a photo of mqtt explorer producing all the readings:
![sofar mqtt explorer photo](https://github.com/cmcgerty/Sofar2mqtt/assets/65698336/caff976b-cf57-43e8-bd6d-0e9b4ca1f0d4)
But the readings are not updating.

**Comment by kiwibhoy** [comment]
[cid:4b1a9661-d050-4a5d-95bc-d36b0e0f2086]
I took this photo today and noticed that "sofar" below Sofar2mqtt looks like another "Topic". Its not in the explorer. Is this normal?

Regards


Mike

________________________________

**Comment by cliffdude** [comment]
The only thing I see there, as you already mentioned is that you are using
2 topics in your config file sofar2mqtt/# to set the modes and the sofar/#
to get the readings.
In MQTT explorer you also show "sofar2mqtt/solarPV1" with data?
I am not using the latest version as I created my own fork and removed some
of the values I don't find relevant.
 The 2 topics may be normal now but you may want to reflash your ESP with a
clean copy of the code.
I have the HYD-4000-ES by the way, it is identical to yours less 1kw output
on the AC.


A quinta, 2/11/2023, 06:39, kiwibhoy ***@***.***> escreveu:

> [cid:4b1a9661-d050-4a5d-95bc-d36b0e0f2086]
> I took this photo today and noticed that "sofar" below Sofar2mqtt looks
> like another "Topic". Its not in the explorer. Is this normal?
>
> Regards
>
>
> Mike
>
> ________________________________
> From: cliffdude ***@***.***>
> Sent: Thursday, 2 November 2023 12:39 am
> To: cmcgerty/Sofar2mqtt ***@***.***>
> Cc: kiwibhoy ***@***.***>; Author ***@***.***>
> Subject: Re: [cmcgerty/Sofar2mqtt] Sofar dxoverdy dashboard occassionally
> shows values (Discussion #71)
>
>
> Looks like a problem with how you setup your MQTT entities.
> The ESP does not sent all values in every message so what you see in MQTT
> explorer is normal, it can only send what it has received and decoded from
> the RS485 and I have to assume that the inverter does not send all data
> every time.
> There is code for HA in the repository, I would suggest creating sensors
> with that code and testing again. Once you have the first valid message
> the
> value of the sensor will persist in HA until the next MQTT message arrives
> with a new value for that variable.
>

**Comment by kiwibhoy** [comment]
Hi,

I reflashed the esp and still have the same problem, your comment re sofar/#, I must of somehow did this when I was cleaning out topics in the App when changing over to HA version of explorer.

But it is interesting that it picked up as a topic when the only place it was put as a topic is in the explorer app.

when Sofar did appear it produced a list and the dashboard had values, they didn't update mind you.

I am using the dxoverdy Hybrid inverter YMAL file and also the dashboard YMAL file from the same source.

The photo below shows that the RS485 data is streaming through and is showing over 400 messages, those messages I think, should be coming out in the form of a list and the list should update the dashboard values every 10 seconds if the value has changed since the last update.
Do you agree?

Refer also to the mqtt add on log you will see the following IP address:

172.30.32.2:46758 on port 1883 appears continually.
Is this normal?

What system are you using to collect the RS485 data?

FYI and nothing to do with the issue:

I've included a photo of my system which you will see has two inverters the Hybrid and a grid tied inverter and two batteries.
The data coming out of the Hybrid is not total power, but the import and export data are total.
I want to be able to make best use of the power generated and reduce the export power if I can put it to use.

Thanks for taking the time to try to help me get this going, is there anything that I can send you that would help make your task easier and make sure I dont waste your time.

Regards

Mike




Regards


Mike

________________________________

**Comment by cliffdude** [comment]
I think you are over complicating this. 
The problem is either the ESP no sending info to HA or HA not interpreting the info correctly and thus not updating sensors etc...
Forget mqtt explorer and all the rest of it and in HA go to Settings -> Devices and services -> mqtt -> configure.
Under "listen to a topic" put 'Sofar2mqtt/#' or whatever you set in the ESP arduino .ino file as the device name then press start listening and wait. After a few seconds you should see a JSON string like this:
PS: If you can't see anything just put a # in the topic to see all MQTT messages, I don't know if you have other devices.

`Message 1 received on Sofar2mqtt/state at 8:10 AM:

{
    "running_state": 2,
    "grid_voltage": 2407,
    "grid_current": 297,
    "grid_freq": 5001,
    "grid_power": 0,
    "battery_power": 53,
    "battery_voltage": 489,
    "battery_current": 1045,
    "systemIO_power": 20,
    "batterySOC": 19,
    "battery_temp": 19,
    "battery_cycles": 170,
    "consumption": 18,
    "solarPV": 71,
    "solarPVAmps": 0,
    "today_generation": 24,
    "today_exported": 0,
    "today_purchase": 433,
    "Solarpv1": 63,
    "Solarpv2": 9,
    "today_consumption": 451,
    "inverter_temp": 32,
    "inverter_HStemp": 20
}`

My mqtt JSON does not have all the values as I have modified Colin's code to only send me what I need. This strings will keep coming in every 10 seconds or so with the updated values. If this works correctly and the values are coming in, not every value must be sent every time, the ESP is doing it's job as intended. In that case the problem is the YAML files in HA.

**Comment by kiwibhoy** [comment]
This is what I get from listening to sofar2mqtt it is not in a list form like yours, although I did have a list once during this exercise with about 10 items in it and I could see that it was displayed in my dashboard but not updating. The results are similar to using mqtt explorer.
I compared my .ino file with the mcgerty one and it is the same.
Also refer the Config YMAL file I am using from dxoverdy.
Does your mqtt log display similar patterns to the one I sent the other day with regard to

172.30.32.2:46758 on port 1883 appears continually.
Is this normal?

[https://res.cdn.office.net/assets/mail/file-icon/png/txt_16x16.png]Config YMAL.rtf<https://1drv.ms/u/s!AssGCVRNcaOTgugl4a0t6N4FLtMoZg>

I am trying hard not to complicate things but this is quite complicated stuff for a 81 year old, ha ha

Regards


Mike

________________________________

**Comment by kiwibhoy** [comment]
Hi 
I got this list today about 4 or 5 intervals one after the other and then it reverted back to just a couple of sensors each interval. I am not changing anything or touching any equipment and I am beginning to wonder if the format of the Inverter RS485 output is changing. Sounds silly I know.



Listening to
Sofar2mqtt/#
Message 4178 received on Sofar2mqtt/state at 1:24 PM:
"running_state": 2,
"grid_voltage": 2357,
"grid_current": 725,
"grid_freq": 4995,
"grid_power": 111,
"battery_power": 286,
"battery_voltage": 543,
"battery_current": 5353,
"systemIO_power": 165,
"batterySOC": 87,
"battery_temp": 25,
"consumption": 67,
"solarPV": 521,
"solarPVAmps": 0,
"today_generation": 672,
"Solarpv1": 191,
"today_consumption": 575,
"inverter_HStemp": 59
Qos
STOP LISTENING

**Comment by cliffdude** [comment]
Ok in that case it could be a comunication error on the rs485. Try a
different power adapter fo the Esp, I had an issue like that once as well.
Also use some twisted pair like cat5 or cat6 to connect he esp to the
inverter rs485.

A domingo, 19/11/2023, 01:39, kiwibhoy ***@***.***> escreveu:

**Reply by kiwibhoy** [reply]
Hi,
I've tried changing the power supply to a couple of others I use and it made no difference.
I had used a twisted pair before so this time I decided to use a twisted pair with screen and earthed down one end of the screen, still no luck.
Its strange how it changes, the number of sensors being read at one time increases each reading until all are present and for the next minute all readings are present. Then goes back to a couple or possibly three readings.
Could you possibly send me a copy of your .ino file so I can see if there are any differences?
Other than that I am at a loss to think of any thing else to do.
Regards Mike

**Reply by kiwibhoy** [reply]
Forgot to say the only time I see "running state":2 is when the whole list is displayed
